### ISSUE TITLE
In spmd training of multiple machines, xp.trace is problematic

### ISSUE THREAD

**Issue by mars1248** [pytorch/xla#7890]
## ❓ Questions and Help
I printed all the thunk that was executed and found that there were a lot of thunk that didn't appear in my tensorboard. And the order of the front and back is also wrong.
I trace according to this example：https://github.com/pytorch/xla/blob/master/test/spmd/test_train_spmd_imagenet.py#L318-L333
xla_version: latest
device: 2 * 8 A100

**Comment by JackCaoG** [comment]
Did you run with `XLA_IR_DEBUG=1 XLA_HLO_DEBUG=1`?

**Comment by mars1248** [comment]
> Did you run with `XLA_IR_DEBUG=1 XLA_HLO_DEBUG=1`?
 yes,Is there any problem with that?

**Comment by JackCaoG** [comment]
no, you should run with those 2 env vars. Does this issue happens when you only use a single machine? I have never run into problems where `xp.Trace` does not show up. I noticed that some times the device and host traces timeline is slightly shifted but the order is always correct.

**Comment by ManfeiBai** [comment]
hi, @JackCaoG, is that ok to assign this ticket to you?
